doc/api/wasi.md

```diff
@@ -0,0 +1,90 @@
+# WebAssembly System Interface (WASI)
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+The WASI API provides an implementation of the [WebAssembly System Interface][]
+specification. WASI gives sandboxed WebAssembly applications access to the
+underlying operating system via a collection of POSIX-like functions.
+
+```js
+'use strict';
+const fs = require('fs');
+const { WASI } = require('wasi');
+const wasi = new WASI({
+  args: process.argv,
+  env: process.env,
+  preopens: {
+    '/sandbox': '/some/real/path/that/wasm/can/access'
+  }
+});
+const importObject = { wasi_unstable: wasi.wasiImport };
+
+(async () => {
+  const wasm = await WebAssembly.compile(fs.readFileSync('./binary.wasm'));
+  const { instance } = await WebAssembly.instantiate(wasm, importObject);
```

Comment:
`WebAssembly.instantiate` returns `Promise<WebAssembly.Instance>` if its first argument is `WebAssembly.Module`.
```suggestion
  const instance = await WebAssembly.instantiate(wasm, importObject);
```